commons/zenoh-codec/src/core/zbuf.rs

```diff
@@ -164,9 +165,8 @@ mod shm {
                                     reader.read_zslices(len, |s| zbuf.push_zslice(s))?;
                                 }
                                 SHM_PTR => {
-                                    let mut zslice: ZSlice = self.codec.read(&mut *reader)?;
-                                    zslice.kind = ZSliceKind::ShmPtr;
-                                    zbuf.push_zslice(zslice);
+                                    let vec: Vec<u8> = self.codec.read(&mut *reader)?;
```

Comment:
We should read a `ZSlice` here and not a `Vec<u8>` which always causes an allocation and a copy.
This is something that should be avoided as much as possible. 
In short, while this PR reduces the size on the stack of the `ZSlice`, it introduces an allocation for every `ShmBufInfoBuffer`.

Reply:
Indeed, forget about that again... I'm not sure there is a simple way to handle that. In fact the issue is our two-phase deserialization, first into a `ZSlice` containing a `ShmBufInfo`, then into  a ZSlice containing a SHM buffer.
Why don't we do both at once?

Reply:
We can definitely define a specific codec (e.g. `ZenohZSliceSHM`) to directly decode a `ZSlice` containing a  `ShmBufInfoBuffer`. The code would look like something like:

```rust
{
   ...
   SHM_PTR => {
      let shm_codec = ZenohZSliceSHM::new();
      let zslice: ZSlice = shm_codec.read(&mut *reader)?;
      zbuf.push_zslice(zslice);
   }
}

impl<R> RCodec<ZSlice, &mut R> for ZenohZSliceSHM
where
    R: Reader,
{
   // Do everything all-in-one
}
```